nomad/structs/job.go

```diff
@@ -16,6 +16,46 @@ const (
 	JobServiceRegistrationsRPCMethod = "Job.GetServiceRegistrations"
 )
 
+type UIJob struct {
+	NamespacedID
+	Name          string
+	Type          string
+	NodePool      string
+	Datacenters   []string
+	Priority      int
+	Allocs        []JobStatusAlloc
+	SmartAlloc    map[string]int
+	GroupCountSum int
+	ChildStatuses []string
+	DeploymentID  string
```

Comment:
For clarity, maybe: 

```suggestion
	LatestDeploymentID  string
```

Also, should this object have any of the indexes for the jobs?

Reply:
I think this is `DeploymentID` because it's meant as an indicator of "Is there currently a deployment happening?"

Maybe `ActiveDeploymentID` in that case, though?

Reply:
I dig `ActiveDeploymentID` :+1:  or I could make it a `bool` `ActiveDeployment`, or similar?  Phil do you imagine that there will actually be any use for the ID, now or later?

for the indexes, do you mean their `ModifyIndex`?  I'm happy to include that or any other fields on here, but so far have been trying to keep it to "only what Phil needs" because it's easy to add stuff later, harder to take away if users start using it for their own purposes.

Reply:
> I dig ActiveDeploymentID 👍 or I could make it a bool ActiveDeployment, or similar? Phil do you imagine that there will actually be any use for the ID, now or later?

I'd have it be the ID because if it's at all relevant that there _is_ a deployment, we're going to want to go looking for information about it anyways. That'll save us an API call.

> for the indexes, do you mean their ModifyIndex? I'm happy to include that or any other fields on here, but so far have been trying to keep it to "only what Phil needs" because it's easy to add stuff later, harder to take away if users start using it for their own purposes.

Yes, but that's totally sensible

Reply:
> Phil do you imagine that there will actually be any use for the ID, now or later

Yep I could see us wanting that as a quick way to link to a deployment directly!

Reply:
we ended up going with a stubby `LatestDeployment` struct that contains more info, since I have it in hand already in the backend, saving the UI from needing to make a separate call to gather anything at all about the deployment. => [b681e8b](https://github.com/hashicorp/nomad/pull/20130/commits/b681e8b8ac207d1be8cb9bc4cad1086b11475bc1)

```shell
$ nomad operator api /v1/jobs/statuses | jq '.[].LatestDeployment'
```

```json
{
  "AllAutoPromote": false,
  "ID": "dedfb786-33b0-fcb0-10bf-1b6c63954fd0",
  "IsActive": false,
  "JobVersion": 0,
  "RequiresPromotion": false,
  "Status": "successful",
  "StatusDescription": "Deployment completed successfully"
}
```